common/src/main/java/org/keycloak/common/Profile.java

```diff
@@ -73,7 +73,10 @@ public enum Feature {
 
         SCRIPTS("Write custom authenticators using JavaScript", Type.PREVIEW),
 
-        TOKEN_EXCHANGE("Token Exchange Service", Type.PREVIEW),
+        TOKEN_EXCHANGE("Token Exchange Service", Type.PREVIEW, 1),
+        TOKEN_EXCHANGE_STANDARD_V2("Standard Token Exchange version 2", Type.EXPERIMENTAL, 2, Feature.ADMIN_FINE_GRAINED_AUTHZ), // TODO: Switch v2 token exchanges to depend admin-fine-grained-authz-v2
+        TOKEN_EXCHANGE_FEDERATED_V2("Federated Token Exchange for external-internal and internal-external token exchange", Type.EXPERIMENTAL, 2, Feature.ADMIN_FINE_GRAINED_AUTHZ),
+        TOKEN_EXCHANGE_SUBJECT_IMPERSONATION_V2("Subject impersonation Token Exchange", Type.EXPERIMENTAL, 2, Feature.ADMIN_FINE_GRAINED_AUTHZ),
```

Comment:
Although the version is 2 for the new token exchange features, the name is different, so you can start the server like this for example:

```bash
./kc.sh start-dev --features token-exchange,token-exchange-standard,token-exchange-federated,admin-fine-grained-authz
```

I'm not saying this is an error but at least we can increase the order of the new v2 standard factory to 1 to be tried before the v1 version.

Reply:
The isAvailable can also be used to avoid the two features are enabled at the same time

Reply:
@rmartinc Thanks for pointing that. I've updated the PR (added additional commit just for the easier review what changed).

I've used the approach with `order()` . The approach with `isAvailable` in profile might be possible too, but will probably require some updates in the `Profile` class itself (due the `isAvailable` is called before `Profile` is initialized, so it is not easily possible to ask for other enabled features). So rather went the easier path for now :-)